bottlerocket/agents/src/bin/ecs-workload-agent/main.rs

```diff
@@ -0,0 +1,499 @@
+                .task_definition(task_def_arn)
+                .launch_type(LaunchType::Ec2)
+                .send()
+                .await
+                .context(error::TaskRunCreationSnafu)?;
+            let run_task_arns: Vec<String> = run_task_output
+                .tasks()
+                .map(|tasks| {
+                    tasks
+                        .iter()
+                        .filter_map(|task| task.task_arn().map(|arn| arn.to_string()))
+                        .collect()
+                })
+                .unwrap();
+            for task_arn in &run_task_arns {
+                task_arns.push(task_arn.clone());
+            }
+        }
+
+        info!("Waiting for tasks to complete...");
+
+        match tokio::time::timeout(
+            Duration::from_secs(30),
+            wait_for_test_completion(&ecs_client, &self.config.cluster_name, &task_arns),
+        )
+        .await
+        {
+            Ok(results) => results,
+            Err(_) => test_results(&ecs_client, &self.config.cluster_name, &task_arns).await,
+        }
+    }
+
+    /// Terminate the test runs. This is a noop for this agent.
+    async fn terminate(&mut self) -> Result<(), Self::E> {
+        Ok(())
+    }
+}
+
+/// Loops waiting for the workload test tasks to complete.
+async fn wait_for_test_completion(
+    ecs_client: &aws_sdk_ecs::Client,
+    cluster_name: &str,
+    task_arns: &[String],
+) -> Result<TestResults, Error> {
+    loop {
+        let results = test_results(ecs_client, cluster_name, task_arns).await?;
+        if results.outcome == Outcome::Pass {
+            return Ok(results);
+        }
+        tokio::time::sleep(Duration::from_secs(2)).await;
+    }
+}
+
+/// Gets the status from our test tasks and populates a `TestResults` struct with the outcome.
+async fn test_results(
+    ecs_client: &aws_sdk_ecs::Client,
+    cluster_name: &str,
+    task_arns: &[String],
+) -> Result<TestResults, Error> {
+    let tasks = ecs_client
+        .describe_tasks()
+        .cluster(cluster_name)
+        .set_tasks(Some(task_arns.to_vec()))
+        .send()
+        .await
+        .context(error::TaskDescribeSnafu)?
+        .tasks()
+        .map(|tasks| tasks.to_owned())
+        .context(error::NoTaskSnafu)?;
+    let running_count = tasks
+        .iter()
+        .filter(|task| task.last_status() == Some("STOPPED"))
+        .filter(|task| task.stop_code() == Some(&TaskStopCode::EssentialContainerExited))
+        .filter(|task| {
+            task.containers()
+                .unwrap_or_default()
+                .iter()
+                .filter(|container| container.exit_code() != Some(0))
+                .count()
+                == 0
+        })
+        .count() as i32;
+    Ok(TestResults {
+        outcome: if 1 == running_count {
+            Outcome::Pass
+        } else {
+            Outcome::Fail
+        },
+        num_passed: running_count as u64,
+        num_failed: (1 - running_count) as u64,
```

Comment:
I'm unsure what `running_count` represents. Is it either `1` or `0` or is the number of running tasks?

Reply:
It would be the running tasks. In the existing `ecs-test-agent` this number is passed in as part of the config, but it should just be the number of total tasks.